src/encodings/levels.rs

```diff
@@ -107,6 +107,14 @@ impl LevelDecoder {
   }
 
   #[inline]
+  // Set byte array explicitly when start position ('start') and length ('len') are known in advance
+  // Returns number of total bytes set for this decoder (len)
+  pub fn set_data_range(&mut self, data: &ByteBufferPtr, start: usize, len: usize) -> usize {
```

Comment:
nit: instead of having a separate function, you can also add a flag to `set_data` to indicate whether the data size should be read from the first i32 of the input buffer. The current solution is fine though so feel free to pick either one. :)

Reply:
Thanks for suggestion! Would it mean that I had to encode both start position and length? I felt like `set_data` implies that entire byte buffer should be set, and `set_data_range` means that we might set only portion of it and not necessarily from beginning. 

My original idea was having `set_data_range` and calling this function from `set_data`. I will have a look again to see if I can make it work with a flag.

Reply:
Hmm.. on a second thought I think adding a flag may not be that easy. I'm OK with keeping the current solution.